demo/test_demos.py

```diff
@@ -73,3 +73,43 @@ def test_demo(file, scalar_type):
             )
             == 0
         )
+
+
+@pytest.mark.parametrize("scalar_type", ["float64", "float32"])
+def test_demo_nvrtc(scalar_type):
+    """Test generated CUDA code with NVRTC."""
+    file = "Components"
+    opts = f"--scalar_type {scalar_type} --cuda"
+    if not sys.platform.startswith("linux"):
+        pytest.skip(reason="NVRTC support only tested on Linux")
+    else:
+        from nvidia import cuda_nvrtc
+
+        nvrtc_dir = os.path.dirname(os.path.realpath(cuda_nvrtc.__file__))
+        cc = os.environ.get("CC", "cc")
+        extra_flags = (
+            "-std=c17 -Wunused-variable -Werror -fPIC -Wno-error=implicit-function-declaration"
+        )
+        assert os.system(f"cd {demo_dir} && ffcx {opts} {file}.py") == 0
+        assert (
+            os.system(
+                f"cd {demo_dir} && "
+                f"{cc} -I../ffcx/codegeneration "
+                f"{extra_flags} "
+                f"-c {file}.c"
+            )
+            == 0
+        )
+        cxx = os.environ.get("CXX", "c++")
+        assert (
+            os.system(
+                f"cd {demo_dir} && "
+                f"{cxx} -I../ffcx/codegeneration -I{nvrtc_dir}/include -L{nvrtc_dir}/lib"
```

Comment:
Can we build with a higher C++ standard? e.g. 20.

And also -Werror

Reply:
Added -Werror but had issues in CI with C++20